Few tweaks

TrueCarry opened the pull request:
- Not using full packet path in dockerfile, so case sensitive errors don't occur(kriechi/Kriechi)
- More permissive regex, to allow slightly different type of region.(ams3 region in DO Spaces don't have dashes in it - AWS4-HMAC-SHA256 Credential=6K5QTQPYH7JAPSB6YCNH/20210420/ams3/s3/aws4_request)
- call log.Fatal(http.ListenAndServ... inside goroutine function, otherwise it behaves same as without go keyword.
- Add prefix to Prometheus metrics.
- Add env variants to arguments.

Kriechi commented:
Yeah 🎉 👍 

Would you mind adding a small example to the README that uses one or two ENV variables, and refer to the `--help` output for more?

Please remove the comment from the Dockerfile - your change looks good! I think the reason for the current setup were some issues with `GOPATH` - but it seems this now works much simpler.

Good catch on the `go func() {...}` wrapper!

TrueCarry commented:
Updated.
After a bit of testing I've found that it doesn't work for put with stream upload.
```
10000;chunk-signature=f3710e3a54a1c788c6541ccbbb9aaf1cf2e0edfb1805f5e0f23704a033a2b124
����
```
Those things end up in uploaded file.
Quick look at received and proxied request shows that while incoming request has req.Header["X-Amz-Content-Sha256"]=="STREAMING-AWS4-HMAC-SHA256-PAYLOAD", outcoming one has req.Header["X-Amz-Content-Sha256"]==sha256(body).
I've tried to set header to that value, but got signing error. Do you have idea if it's fixable? (I use minio for uploading)

P.S. For some reason for streaming requests minio doesn't add space after "," in auth header, so temporarily I've changed auth check to work with it.
![image](https://user-images.githubusercontent.com/5431520/115523504-cb536880-a295-11eb-88e1-e93b81e5c744.png)

Kriechi commented:
@TrueCarry is this a new issue you are describing, or something that is now broken with this PR?
I'd be happy to merge if this is "as stable as master", and then track this problem on a new GitHub issue.

TrueCarry commented:
This should be stable as master. But in both versions PUT requests not working.
